src/main/java/appeng/integration/modules/crafttweaker/CTModule.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * This file is part of Applied Energistics 2.
+ * Copyright (c) 2013 - 2017, AlgorithmX2, All rights reserved.
+ *
+ * Applied Energistics 2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.integration.modules.crafttweaker;
+
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+import net.minecraft.creativetab.CreativeTabs;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.NonNullList;
+import net.minecraftforge.oredict.OreDictionary;
+
+import crafttweaker.CraftTweakerAPI;
+import crafttweaker.IAction;
+import crafttweaker.api.item.IIngredient;
+import crafttweaker.api.item.IItemStack;
+
+import appeng.integration.abstraction.ICraftTweaker;
+import appeng.util.Platform;
+
+
+public class CTModule implements ICraftTweaker
+{
+	static final List<IAction> MODIFICATIONS = new ArrayList<>();
+
+	@Override
+	public void preInit()
+	{
+		CraftTweakerAPI.registerClass( GrinderRecipes.class );
+		CraftTweakerAPI.registerClass( InscriberRecipes.class );
+	}
+
+	@Override
+	public void postInit()
+	{
+		MODIFICATIONS.forEach( CraftTweakerAPI::apply );
+	}
+
+	public static ItemStack toStack( IItemStack iStack )
+	{
+		if( iStack == null )
+		{
+			return ItemStack.EMPTY;
+		}
+		else
+		{
+			return (ItemStack) iStack.getInternal();
+		}
+	}
+
+	public static List<ItemStack> toStackExpand( IItemStack iStack )
+	{
+		if( iStack == null )
+		{
+			return Collections.emptyList();
+		}
+		else
+		{
+			ItemStack is = (ItemStack) iStack.getInternal();
+			if( !is.isItemStackDamageable() && is.getItemDamage() == OreDictionary.WILDCARD_VALUE )
+			{
+				NonNullList<ItemStack> ret = NonNullList.create();
+				is.getItem().getSubItems( CreativeTabs.SEARCH, ret );
+				return ret.stream().map( i -> new ItemStack( i.getItem(), iStack.getAmount(), i.getItemDamage() ) ).collect( Collectors.toList() );
+			}
+			else
+			{
+				return Collections.singletonList( is );
+			}
+		}
+	}
+
+	public static Optional<Collection<ItemStack>> toStacks( IIngredient ingredient )
+	{
+		if( ingredient == null )
+		{
+			return Optional.empty();
+		}
+		Set<ItemStack> ret = new TreeSet<ItemStack>( CTModule::compareItemStacks );
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2293)